R/shinyform.R

```diff
@@ -79,21 +79,26 @@ loadData <- function(storage) {
   }
 }
 
-
-
 # Takes data from your shinyforms inputs and saves it to a flat file. 
 # Writes form inputs to a storage type and names it using a timestamp.
 # @param data Dataframe taken from input shiny object
-# @param storage A list with variable type defining users perferred type of storage and storage path
+# @param storage A list with variable type defining users perferred type of storage, filename and storage path
 saveDataFlatfile <- function(data, storage) {
-  fileName <- paste0(
-    paste(
-      format(Sys.time(), "%Y%m%d-%H%M%OS"),
-      digest::digest(data, algo = "md5"),
-      sep = "_"
-    ),
-    ".csv"
-  )
+  if (!is.null(storage$filename)) {
```

Comment:
It's more readable to start with the `is.null()` case rather than the `!is.null()`, it's just easier on the brain to do positives than negatives. And since the if statement is really just determining the base filename (not the extension), it's a bit simpler to do this:

```
if (is.null(...) {
  filename <- paste(...)
} else {
  filename <- storage$filename
}
filename <- paste0(filename, ".csv")
```

Reply:
thanks for the feedback